src/index.js

```diff
@@ -1,86 +1,89 @@
-  self.app.post(`${self.baseApi}/all`, notificationController.updateAllUnread);
+    // Expose endpoints
+    self.app.get(`${self.baseApi}/all/:userid`, notificationController.getAll);
+    self.app.put(`${self.baseApi}/:notificationId`, notificationController.updateUnread);
+    self.app.delete(`${self.baseApi}/:notificationId`, notificationController.delete);
+    self.app.post(`${self.baseApi}/all`, notificationController.updateAllUnread);
 
-  self.io.on('connection', (socket) => {
-    socket.emit('who-are-you');
+    self.io.on('connection', (socket) => {
+      socket.emit('who-are-you');
 
-    socket.on('check-in', (user, ack) => {
-      self.io.sockets[user.id] = socket;
-      self.sendUnread(user.id);
+      socket.on('check-in', (user, ack) => {
+        self.io.sockets[user.id] = socket;
+        self.sendUnread(user.id);
 
-      // Send to client endpoint base route forexposed endpoints
-      ack(`{"baseApi":"${self.baseApi}"}`);
+        // Send to client endpoint base route forexposed endpoints
+        ack(`{"baseApi":"${self.baseApi}"}`);
+      });
+
+      socket.on('check-news', (user) => {
+        self.sendUnsent(user.id);
+      });
     });
+  };
 
-    socket.on('check-news', (user) => {
-      self.sendUnsent(user.id);
+
+  send(event, notification) {
+    const user = notification.to;
+    this.io.sockets[user].emit(event, notification, (ack) => {
+      if (ack) {
+        notificationController
+        .update(notification._id, { sent: true }, //eslint-disable-line
+          (err) => {
+            if (err) throw err;
+          });
+      }
     });
-  });
-};
+  }
 
-Notification.prototype.send = (event, notification) => {
-  const user = notification.to;
-  this.io.sockets[user].emit(event, notification, (ack) => {
-    if (ack) {
+  sendUnread(userId) {
+    const self = this;
+    if (userId) {
       notificationController
-      .update(notification._id, { sent: true }, //eslint-disable-line
-        (err) => {
-          if (err) throw err;
+      .getUnread(userId, (err, data) => {
+        if (err) throw err;
+        data.forEach((notification) => {
+          self.send('unread', notification);
         });
+      });
     }
-  });
-};
+  }
 
-Notification.prototype.sendUnread = (userId) => {
-  const self = this;
-  if (userId) {
-    notificationController
-    .getUnread(userId, (err, data) => {
-      if (err) throw err;
-      data.forEach((notification) => {
-        self.send('unread', notification);
+  sendUnsent(userId) {
+    const self = this;
+    if (userId) {
+      notificationController
+      .getUnsent(userId, (err, data) => {
+        if (err) throw err;
+        data.forEach((notification) => {
+          self.send('news', notification);
+        });
       });
-    });
+    }
   }
-};
 
-Notification.prototype.sendUnsent = (userId) => {
-  const self = this;
-  if (userId) {
-    notificationController
-    .getUnsent(userId, (err, data) => {
+  create(type, to, title, msg) {
```

Comment:
Four parameters sounds like too much, do you think it's better if we just pass an object here?